src/Umbraco.Web.Common/Security/ConfigureKestrelServerOptions.cs

```diff
@@ -12,7 +12,10 @@ public class ConfigureKestrelServerOptions : IConfigureOptions<KestrelServerOpti
         public void Configure(KestrelServerOptions options)
         {
             // convert from KB to bytes
-            options.Limits.MaxRequestBodySize = _runtimeSettings.Value.MaxRequestLength.HasValue ? _runtimeSettings.Value.MaxRequestLength.Value * 1024 : long.MaxValue;
+            if (_runtimeSettings.Value.MaxRequestLength.HasValue)
+            {
+                options.Limits.MaxRequestBodySize = _runtimeSettings.Value.MaxRequestLength.Value * 1024;
+            }
```

Comment:
A valid point was made here, should we instead set the default if no settings is specified to the same as it was in V8? I think it was like 50 MB, that way there's no behavioural change when migrating from V8 to V9 kestrel? Of course we sadly can't do this for IIS since it's not possible as you mentioned, but it might be worthwhile to get as close to the V8 experience as possible.

A different angle is that some installs might already be running with Kestrel with no config set, and is relying on the max value being set, and I guess that this would be a behavioural breaking change for them?

Reply:
It is a behavioral change but i think its a bigger security issue if we just default to the maximum of 4GB, will change this to 50MB to align with v8 👍